src/helpers/host-component-names.tsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import { Text, TextInput, View } from 'react-native';
+import TestRenderer from 'react-test-renderer';
+import { configureInternal, getConfig, HostComponentNames } from '../config';
+import { getQueriesForElement } from '../within';
+
+const defaultErrorMessage = `There seems to be an issue with your configuration that prevents React Native Testing Library from working correctly.
+Please check if you are using compatible versions of React Native and React Native Testing Library.`;
+
+export function detectHostComponentNamesIfNeeded(): HostComponentNames {
```

Comment:
Looking at the call-sites of this function, maybe we can simplify it to:
```suggestion
export function getHostComponentNames(): HostComponentNames {
```